ReSharper.FSharp/src/FSharp.Psi.Services/src/Generate/GenerateProvider.fs

```diff
@@ -231,22 +250,20 @@ type FSharpOverridingMembersBuilder() =
 
             if objModelTypeRepr :? IStructRepresentation then objModelTypeRepr :> _ else
 
-            let equalsToken = typeDecl.EqualsToken.NotNull()
-
-            let anchor =
-                let afterComment = getLastMatchingNodeAfter isInlineSpaceOrComment equalsToken
-                let afterSpace = getLastMatchingNodeAfter isInlineSpace equalsToken
-                if afterComment != afterSpace then afterComment else equalsToken :> _
-
-            deleteChildRange anchor.NextSibling typeRepr
-
-            equalsToken :> _
+            objModelTypeRepr
 
         let (anchor: ITreeNode), indent =
+            let getIdentForUnion (unionRepr: IUnionRepresentation) =
+                if typeDecl.StartLine <> unionRepr.StartLine then unionRepr.Indent else
+                typeDecl.Indent + typeDecl.GetIndentSize()
```

Comment:
Are unions specific in any way? What is the difference in terms of generating overrides with, say, records?

```fsharp
type R = { F: int }
```

I think calculating the indent should be done after the representation was moved to a new line and shifted, if needed. Then the logic can probably be more or less unified between the representation kinds.

Reply:
I could simplify the logic a bit, but I'm not sure if that was, what you had in mind.

The difference I see between unions/classes and record types is that we can't use the `typeRepr.Indent` value during the indent calculation for them as that is the value before the move/shift or deletion (in case of classes).

As records always have a Representation, unlike classes, and we don't move that Representation, we can use `typeRepr.Indent` for them. I think, we even _must_ use the `typeRepr.Indent` for records to deal with cases like:
```fsharp
        type MyRecord = { Field1: int; Field2: int }

                        override this.ToString() = failwith "todo"
```
Here, the addition of the override must have the same indent as the representation.

Reply:
I think I'd move any kind of representation that doesn't start on a new line, just like it's done for unions.

Reply:
Makes sense. Should be done now for unions, classes, structs and records.